daemon/daemon.go

```diff
@@ -407,20 +407,12 @@ func (daemon *Daemon) reserveName(id, name string) (string, error) {
 
 		conflictingContainer, err := daemon.GetByName(name)
 		if err != nil {
-			if strings.Contains(err.Error(), "Could not find entity") {
-				return "", err
-			}
```

Comment:
@thaJeztah  On second thought perhaps allowing the user to delete the broken containers is a bit dangerous. The container's config and the record in the graph db can be deleted, but the data generated by the container can not be easily deleted because it's using a different storage driver. If we only delete the index and config of the container when the user run `docker rm xxx`, there'll be garbage data left on the disk.
Is it possible to maintain a `linkgraph.db` for each graph driver?  This way the user can switch between different graph drivers without such problem.
I noticed that each graph driver already has its own image repositories file, such as `repositories-aufs`, `repositories-devicemapper`, etc.

BTW: I created a patch here https://github.com/mountkin/docker/commit/3563666d3d98661385131ef723cb33bfd325410b but I'm hesitating whether it's the right way to handle this situation.